Plugin fails to build with "Unsupported class file major version 63" exception

Extension Name: vscode-gradle
Extension Version: v3.12.5
OS Version: Windows 11
VSCode version: 1.73.0

**Describe the bug**
When the plugin is enabled and a Gradle project builds, the process fails with the following message: `A request has failed. See the output for more information.` Looking at the output shows the logs described below. However, some of the plugins features seems to continue working. The error seems to be related to Java 19, but I confirmed it's also happening with Java 18.

**To Reproduce**
1. Create a Gradle project with Java 18 or 19.
2. Go to the `build.gradle` file.
3. Save the file to trigger the build.
4. Wait for the error to appear.

**Expected behavior**
The project builds with no errors.

**Screenshots**
![image](https://user-images.githubusercontent.com/3087228/199650455-fc2771df-8ef7-4d40-a0f7-0e27dd0d505a.png)

**Output from "Gradle for Java"**
<details>
  <summary>Click to see output</summary>

```
Nov 02, 2022 11:57:39 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: java.lang.reflect.InvocationTargetException
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:577)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1589)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:119)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	... 11 more
Caused by: BUG! exception in phase 'semantic analysis' in source unit 'file:///c%3A/Repos/orkestra/orkestra-server/build.gradle' Unsupported class file major version 63
	at org.codehaus.groovy.control.CompilationUnit$ISourceUnitOperation.doPhaseOperation(CompilationUnit.java:943)
	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:642)
	at com.microsoft.gradle.GradleServices.compile(GradleServices.java:178)
	at com.microsoft.gradle.GradleServices.recompileAll(GradleServices.java:195)
	at com.microsoft.gradle.GradleServices.executeCommand(GradleServices.java:379)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	... 13 more
Caused by: java.lang.IllegalArgumentException: Unsupported class file major version 63
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:196)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:177)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:163)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:284)
	at org.codehaus.groovy.ast.decompiled.AsmDecompiler.parseClass(AsmDecompiler.java:81)
	at org.codehaus.groovy.control.ClassNodeResolver.findDecompiled(ClassNodeResolver.java:251)
	at org.codehaus.groovy.control.ClassNodeResolver.tryAsLoaderClassOrScript(ClassNodeResolver.java:189)
	at org.codehaus.groovy.control.ClassNodeResolver.findClassNode(ClassNodeResolver.java:169)
	at org.codehaus.groovy.control.ClassNodeResolver.resolveName(ClassNodeResolver.java:125)
	at org.codehaus.groovy.control.ResolveVisitor.resolveToOuter(ResolveVisitor.java:840)
	at org.codehaus.groovy.control.ResolveVisitor.resolve(ResolveVisitor.java:425)
	at org.codehaus.groovy.control.ResolveVisitor.resolveFromDefaultImports(ResolveVisitor.java:593)
	at org.codehaus.groovy.control.ResolveVisitor.resolveFromDefaultImports(ResolveVisitor.java:557)
	at org.codehaus.groovy.control.ResolveVisitor.resolve(ResolveVisitor.java:424)
	at org.codehaus.groovy.control.ResolveVisitor.resolve(ResolveVisitor.java:387)
	at org.codehaus.groovy.control.ResolveVisitor.resolveOrFail(ResolveVisitor.java:349)
	at org.codehaus.groovy.control.ResolveVisitor.resolveOrFail(ResolveVisitor.java:341)
	at org.codehaus.groovy.control.ResolveVisitor.resolveOrFail(ResolveVisitor.java:337)
	at org.codehaus.groovy.control.ResolveVisitor.transformConstructorCallExpression(ResolveVisitor.java:1220)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:871)
	at org.codehaus.groovy.control.ResolveVisitor.transformDeclarationExpression(ResolveVisitor.java:1282)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:863)
	at org.codehaus.groovy.ast.ClassCodeExpressionTransformer.visitExpressionStatement(ClassCodeExpressionTransformer.java:108)
	at org.codehaus.groovy.ast.stmt.ExpressionStatement.visit(ExpressionStatement.java:40)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:86)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:216)
	at org.codehaus.groovy.control.ResolveVisitor.visitBlockStatement(ResolveVisitor.java:1574)
	at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:69)
	at org.codehaus.groovy.ast.ClassCodeExpressionTransformer.visitIfElse(ClassCodeExpressionTransformer.java:120)
	at org.codehaus.groovy.ast.stmt.IfStatement.visit(IfStatement.java:41)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:86)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:216)
	at org.codehaus.groovy.control.ResolveVisitor.visitBlockStatement(ResolveVisitor.java:1574)
	at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:69)
	at org.codehaus.groovy.control.ResolveVisitor.transformClosureExpression(ResolveVisitor.java:1208)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:869)
	at org.codehaus.groovy.ast.expr.Expression.transformExpressions(Expression.java:49)
	at org.codehaus.groovy.ast.expr.ArgumentListExpression.transformExpression(ArgumentListExpression.java:66)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:876)
	at org.codehaus.groovy.control.ResolveVisitor.transformMethodCallExpression(ResolveVisitor.java:1256)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:867)
	at org.codehaus.groovy.ast.ClassCodeExpressionTransformer.visitExpressionStatement(ClassCodeExpressionTransformer.java:108)
	at org.codehaus.groovy.ast.stmt.ExpressionStatement.visit(ExpressionStatement.java:40)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:86)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:216)
	at org.codehaus.groovy.control.ResolveVisitor.visitBlockStatement(ResolveVisitor.java:1574)
	at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:69)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitClassCodeContainer(ClassCodeVisitorSupport.java:165)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitConstructorOrMethod(ClassCodeVisitorSupport.java:138)
	at org.codehaus.groovy.ast.ClassCodeExpressionTransformer.visitConstructorOrMethod(ClassCodeExpressionTransformer.java:66)
	at org.codehaus.groovy.control.ResolveVisitor.visitConstructorOrMethod(ResolveVisitor.java:305)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitMethod(ClassCodeVisitorSupport.java:133)
	at org.codehaus.groovy.ast.ClassNode.visitMethods(ClassNode.java:1163)
	at org.codehaus.groovy.ast.ClassNode.visitContents(ClassNode.java:1156)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitClass(ClassCodeVisitorSupport.java:56)
	at org.codehaus.groovy.control.ResolveVisitor.visitClass(ResolveVisitor.java:1485)
	at org.codehaus.groovy.control.ResolveVisitor.startResolving(ResolveVisitor.java:264)
	at org.codehaus.groovy.control.CompilationUnit.lambda$0(CompilationUnit.java:756)
	at org.codehaus.groovy.control.CompilationUnit$ISourceUnitOperation.doPhaseOperation(CompilationUnit.java:934)
	... 18 more

[Error - 11:57:39 PM] Request workspace/executeCommand failed.
  Message: Internal error.
  Code: -32603 
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:577)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1589)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:119)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	... 11 more
Caused by: BUG! exception in phase 'semantic analysis' in source unit 'file:///c%3A/Repos/orkestra/orkestra-server/build.gradle' Unsupported class file major version 63
	at org.codehaus.groovy.control.CompilationUnit$ISourceUnitOperation.doPhaseOperation(CompilationUnit.java:943)
	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:642)
	at com.microsoft.gradle.GradleServices.compile(GradleServices.java:178)
	at com.microsoft.gradle.GradleServices.recompileAll(GradleServices.java:195)
	at com.microsoft.gradle.GradleServices.executeCommand(GradleServices.java:379)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	... 13 more
Caused by: java.lang.IllegalArgumentException: Unsupported class file major version 63
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:196)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:177)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:163)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:284)
	at org.codehaus.groovy.ast.decompiled.AsmDecompiler.parseClass(AsmDecompiler.java:81)
	at org.codehaus.groovy.control.ClassNodeResolver.findDecompiled(ClassNodeResolver.java:251)
	at org.codehaus.groovy.control.ClassNodeResolver.tryAsLoaderClassOrScript(ClassNodeResolver.java:189)
	at org.codehaus.groovy.control.ClassNodeResolver.findClassNode(ClassNodeResolver.java:169)
	at org.codehaus.groovy.control.ClassNodeResolver.resolveName(ClassNodeResolver.java:125)
	at org.codehaus.groovy.control.ResolveVisitor.resolveToOuter(ResolveVisitor.java:840)
	at org.codehaus.groovy.control.ResolveVisitor.resolve(ResolveVisitor.java:425)
	at org.codehaus.groovy.control.ResolveVisitor.resolveFromDefaultImports(ResolveVisitor.java:593)
	at org.codehaus.groovy.control.ResolveVisitor.resolveFromDefaultImports(ResolveVisitor.java:557)
	at org.codehaus.groovy.control.ResolveVisitor.resolve(ResolveVisitor.java:424)
	at org.codehaus.groovy.control.ResolveVisitor.resolve(ResolveVisitor.java:387)
	at org.codehaus.groovy.control.ResolveVisitor.resolveOrFail(ResolveVisitor.java:349)
	at org.codehaus.groovy.control.ResolveVisitor.resolveOrFail(ResolveVisitor.java:341)
	at org.codehaus.groovy.control.ResolveVisitor.resolveOrFail(ResolveVisitor.java:337)
	at org.codehaus.groovy.control.ResolveVisitor.transformConstructorCallExpression(ResolveVisitor.java:1220)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:871)
	at org.codehaus.groovy.control.ResolveVisitor.transformDeclarationExpression(ResolveVisitor.java:1282)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:863)
	at org.codehaus.groovy.ast.ClassCodeExpressionTransformer.visitExpressionStatement(ClassCodeExpressionTransformer.java:108)
	at org.codehaus.groovy.ast.stmt.ExpressionStatement.visit(ExpressionStatement.java:40)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:86)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:216)
	at org.codehaus.groovy.control.ResolveVisitor.visitBlockStatement(ResolveVisitor.java:1574)
	at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:69)
	at org.codehaus.groovy.ast.ClassCodeExpressionTransformer.visitIfElse(ClassCodeExpressionTransformer.java:120)
	at org.codehaus.groovy.ast.stmt.IfStatement.visit(IfStatement.java:41)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:86)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:216)
	at org.codehaus.groovy.control.ResolveVisitor.visitBlockStatement(ResolveVisitor.java:1574)
	at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:69)
	at org.codehaus.groovy.control.ResolveVisitor.transformClosureExpression(ResolveVisitor.java:1208)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:869)
	at org.codehaus.groovy.ast.expr.Expression.transformExpressions(Expression.java:49)
	at org.codehaus.groovy.ast.expr.ArgumentListExpression.transformExpression(ArgumentListExpression.java:66)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:876)
	at org.codehaus.groovy.control.ResolveVisitor.transformMethodCallExpression(ResolveVisitor.java:1256)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:867)
	at org.codehaus.groovy.ast.ClassCodeExpressionTransformer.visitExpressionStatement(ClassCodeExpressionTransformer.java:108)
	at org.codehaus.groovy.ast.stmt.ExpressionStatement.visit(ExpressionStatement.java:40)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:86)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:216)
	at org.codehaus.groovy.control.ResolveVisitor.visitBlockStatement(ResolveVisitor.java:1574)
	at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:69)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitClassCodeContainer(ClassCodeVisitorSupport.java:165)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitConstructorOrMethod(ClassCodeVisitorSupport.java:138)
	at org.codehaus.groovy.ast.ClassCodeExpressionTransformer.visitConstructorOrMethod(ClassCodeExpressionTransformer.java:66)
	at org.codehaus.groovy.control.ResolveVisitor.visitConstructorOrMethod(ResolveVisitor.java:305)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitMethod(ClassCodeVisitorSupport.java:133)
	at org.codehaus.groovy.ast.ClassNode.visitMethods(ClassNode.java:1163)
	at org.codehaus.groovy.ast.ClassNode.visitContents(ClassNode.java:1156)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitClass(ClassCodeVisitorSupport.java:56)
	at org.codehaus.groovy.control.ResolveVisitor.visitClass(ResolveVisitor.java:1485)
	at org.codehaus.groovy.control.ResolveVisitor.startResolving(ResolveVisitor.java:264)
	at org.codehaus.groovy.control.CompilationUnit.lambda$0(CompilationUnit.java:756)
	at org.codehaus.groovy.control.CompilationUnit$ISourceUnitOperation.doPhaseOperation(CompilationUnit.java:934)
	... 18 more
```
</details>

**Does the bug still exist if you disable all other extensions?**
Yes

**Additional context**
The issues happens in the following scenarios:
- Java 18 - Gradle v7.5.1
- Java 19 - Gradle v7.6-rc-1

**PS:** The project builds correctly on a terminal (e.g. running `./gradlew build`) on both scenarios

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -55,7 +55,7 @@ jobs:
       fail-fast: false
       matrix:
         node-version: [16.14.2]
-        java-version: ["8", "11", "17"]
+        java-version: ["8", "11", "17", "19"]
         os: [ubuntu-latest, windows-latest, macos-latest]
     steps:
       - uses: actions/checkout@v2

--- a/README.md
+++ b/README.md
@@ -9,7 +9,7 @@ This VS Code extension provides a visual interface for your Gradle build. You ca
 ## Requirements
 
 - [VS Code >= 1.66.0](https://code.visualstudio.com/download)
-- [Java from 8 to 17](https://adoptopenjdk.net/)
+- [Java from 8 to 19](https://adoptopenjdk.net/)
 
 ## Project Discovery
 

--- a/extension/src/util/config.ts
+++ b/extension/src/util/config.ts
@@ -36,7 +36,7 @@ export async function getSupportedJavaHome(): Promise<string | undefined> {
         const runtime = await getRuntime(javaHome, { withVersion: true });
         if (runtime?.version) {
             // check the JDK version of given java home is supported, otherwise return undefined
-            return runtime.version.major >= 8 && runtime.version.major <= 17 ? javaHome : undefined;
+            return runtime.version.major >= 8 && runtime.version.major <= 19 ? javaHome : undefined;
         }
     }
     return undefined;

--- a/gradle-language-server/build.gradle
+++ b/gradle-language-server/build.gradle
@@ -10,6 +10,9 @@ java {
 
 repositories  {
   mavenCentral()
+  maven {
+    url "https://groovy.jfrog.io/artifactory/plugins-release"
+  }
 }
 
 application {
@@ -20,7 +23,7 @@ application {
 dependencies {
   implementation "org.eclipse.lsp4j:org.eclipse.lsp4j:0.12.0"
   implementation "org.eclipse.lsp4j:org.eclipse.lsp4j.jsonrpc:0.12.0"
-  implementation "org.codehaus.groovy:groovy-eclipse-batch:3.0.8-01"
+  implementation "org.codehaus.groovy:groovy-eclipse-batch:4.0.9-02"
   implementation "com.google.code.gson:gson:2.8.9"
   implementation "org.apache.bcel:bcel:6.6.1"
   testImplementation "org.junit.jupiter:junit-jupiter-api:5.8.1"

--- a/gradle-server/src/main/java/com/github/badsyntax/gradle/utils/Utils.java
+++ b/gradle-server/src/main/java/com/github/badsyntax/gradle/utils/Utils.java
@@ -18,8 +18,11 @@ public class Utils {
 
 	private static Version getLowestGradleVersion(Version javaVersion) {
 		// Ref: https://docs.gradle.org/current/userguide/compatibility.html
-		if (javaVersion.isAtLeast("17")) {
-			// See: https://docs.gradle.org/7.3-rc-3/release-notes.html#java17
+		if (javaVersion.isAtLeast("19")) {
+			return new Version("7.6");
+		} else if (javaVersion.isAtLeast("18")) {
+			return new Version("7.5");
+		} else if (javaVersion.isAtLeast("17")) {
 			return new Version("7.3");
 		} else if (javaVersion.isAtLeast("16")) {
 			return new Version("7.0");

--- a/gradle-server/src/test/java/com/github/badsyntax/gradle/UtilsTest.java
+++ b/gradle-server/src/test/java/com/github/badsyntax/gradle/UtilsTest.java
@@ -13,6 +13,15 @@ import org.junit.Test;
 public class UtilsTest {
 
 	@Test
+	public void testJava19Compatibility() {
+		Version javaVersion = new Version("19.0.1");
+		Version gradleVersion = new Version("7.6.0");
+		assertFalse(Utils.hasCompatibilityError(gradleVersion, javaVersion));
+		gradleVersion = new Version("7.5.0");
+		assertTrue(Utils.hasCompatibilityError(gradleVersion, javaVersion));
+	}
+
+	@Test
 	public void testJava17Compatibility() {
 		Version javaVersion = new Version("17.0.1");
 		Version gradleVersion = new Version("7.3.0-rc2");

